baseplate/diagnostics/tracing.py

```diff
@@ -223,6 +228,43 @@ def record(self):
         self.recorder.send(self)
 
 
+class TraceLocalSpanObserver(TraceSpanObserver):
```

Comment:
If in the future we want to create a new type of observer how do we decide whether to make a completely new class like this vs. just updating `on_child_span_created` like in `MetricsServerSpanObserver`? Is this separate class just needed here because traces/spans are so coupled?